File: parksathi-android/app/src/main/java/in/parksathi/app/dto/Booking.kt
package `in`.parksathi.app.dto

import com.google.gson.annotations.SerializedName

enum class BookingStatus {
    @SerializedName("pending") PENDING,
    @SerializedName("confirmed") CONFIRMED,
    @SerializedName("cancelled") CANCELLED,
    @SerializedName("completed") COMPLETED,
    @SerializedName("acquired") ACQUIRED
}

data class BookingResponse(
    @SerializedName("uid") val uid: String,
    @SerializedName("booking_id") val bookingId: String,
    @SerializedName("parking_id") val parkingId: String,
    @SerializedName("booking_status") val bookingStatus: BookingStatus?,
    @SerializedName("parking_name") val parkingName: String,
    @SerializedName("acquired_at") val acquiredAt: String? = null
)

File: parksathi-backend/app/schema/booking.py
from pydantic import BaseModel
from datetime import datetime
from app.model.booking import BookingStatus
from uuid import UUID

class BookingResponse(BaseModel):
    uid: str
    booking_id: UUID
    parking_id: UUID
    booking_status: BookingStatus
    parking_name: str
    acquired_at: datetime | None = None
    
class OwnerBookingResponse(BaseModel):
    user_name: str
    booking_id: UUID
    booking_status: BookingStatus
    created_at: datetime
    acquired_at: datetime | None = None
    completed_at: datetime | None = None